safe-global/web-core
# fix: prevent scroll adjusting number inputs

## What it solves

Resolves #1234

## How this PR fixes it

A new `NumberField` component has been created (isolating the common `type` and `autoComplete` attributes). It blurs the field `onWheel`.

A `ref` is used as `TextField` doesn't work with `currentTarget`.

## How to test it

Ensure the following fields work as expected, with validation and preventing scroll from editing the value:

- Transaction filter: amount, nonce
- Spending limit amount
- Send asset amount
- Advanced transaction parameters: nonce, safeTxGas, wallet nonce, gas limit, maxPriorityFeePerGas, maxFeePerGas
